src/System.Net.Http/src/System/Net/Http/HttpContent.cs

```diff
@@ -142,7 +142,9 @@ internal void SetBuffer(byte[] buffer, int offset, int count)
 
         internal bool TryGetBuffer(out ArraySegment<byte> buffer)
         {
-            return _bufferedContent != null && _bufferedContent.TryGetBuffer(out buffer);
+            bool result = _bufferedContent != null && _bufferedContent.TryGetBuffer(out buffer);
+            Debug.Assert(result, "Performance: TryGetBuffer failed, which means that reads from the content will copy the data!");
```

Comment:
I'm not sure asserting is the right thing to do here.  This is more of something we might want to log. But asserting implies some logic is broken and that isn't true.